eq-author/src/components/EditorLayout/Header/index.test.js

```diff
@@ -92,14 +93,33 @@ describe("Header", () => {
     expect(getByText(questionnaire.displayName)).toBeTruthy();
   });
 
-  it("should show a link for previewing", () => {
-    const { getByText, getByTestId } = renderWithContext(<Header {...props} />);
+  describe("view survey button", () => {
+    it("should show a button for previewing questionnaire", () => {
+      const { getByText, getByTestId } = renderWithContext(
+        <Header {...props} />
+      );
 
-    const link = getByTestId("btn-preview");
-    expect(link.getAttribute("href")).toMatch(
-      new RegExp(`/launch/${questionnaire.id}$`)
-    );
-    expect(getByText("View survey")).toBeTruthy();
+      const viewSurveyButton = getByTestId("btn-preview");
+      expect(viewSurveyButton.getAttribute("href")).toMatch(
+        new RegExp(`/launch/${questionnaire.id}$`)
+      );
+      expect(getByText("View survey")).toBeTruthy();
+    });
+
+    it("should disable the view survey button when questionnaire is invalid", () => {
+      questionnaire.totalErrorCount = 1;
+      const { getByTestId } = renderWithContext(<Header {...props} />);
+
+      const viewSurveyButton = getByTestId("btn-preview");
+      expect(viewSurveyButton.getAttribute("disabled")).toBe("");
+    });
+
+    it("should not disable the view survey button when questionnaire is valid", () => {
+      const { getByTestId } = renderWithContext(<Header {...props} />);
+
+      const viewSurveyButton = getByTestId("btn-preview");
+      expect(viewSurveyButton.getAttribute("disabled")).toBeFalsy();
```

Comment:
Empty string is also falsy. So we can check for something explicitly like `null` or `undefined`

Reply:
Aah thats right, I changed it to check for `null`.

Reply:
Might it be worth bringing in something like https://github.com/testing-library/jest-dom to assert the button is disabled. This way of doing it doesn't feel very rtl-ly if that makes sense?

Reply:
That sounds like a much better idea. Good find.

Reply:
Now checks using this. And fixed the button so you no longer can tab into it.

Reply:
Whats wrong with being able to tab to it? https://ux.stackexchange.com/questions/103239/should-disabled-elements-be-focusable-for-accessibility-purposes

Reply:
The user was able to tab into it and click enter to still go to the link, even when it was disabled. Hamish suggests not being able to tab into it. And there is no point being able to tab into something you cant interact with.